## Calculating the Beta of a Stock

*Suggested Answers follow (usually there are multiple ways to solve a problem in Python).*

Obtain data for Microsoft and S&P 500 for the period 1st of January 2012 – 31st of December 2016 from Yahoo Finance.

In [ ]:

import numpy as np
import pandas as pd
import yfinance as yf
from pandas_datareader import data as wb

tickers = ['MSFT', '^GSPC']
data = pd.DataFrame()

for a in tickers:
    data[a] = yf.download(a, start='2012-1-1', end='2016-12-31', auto_adjust=False)['Adj Close']

[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed


Let S&P 500 act as the market.

*****

Calculate the beta of Microsoft.

Tính tỷ suất sinh lợi:

In [ ]:
sec_returns = np.log( data / data.shift(1) )

Tính Tử số (Covariance):

In [ ]:
cov = sec_returns.cov() * 250
cov

MSFT     ^GSPC
MSFT   0.053781  0.018213
^GSPC  0.018213  0.016362

In [ ]:
cov_with_market = cov.iloc[0,1]
cov_with_market

np.float64(0.018212787511827776)

Tính Mẫu số (Variance):

tính phương sai hàng ngày của thị trường (.var()) rồi nhân với 250 để ra phương sai năm.

market_var (kết quả là 0.0163) chính là phương sai năm của ^GSPC.

In [ ]:
market_var = sec_returns['^GSPC'].var() * 250
market_var

0.016361631002308474

** Beta: **
### $$
\beta_{pg} = \frac{\sigma_{pg,m}}{\sigma_{m}^2}
$$

In [ ]:
MSFT_beta = cov_with_market / market_var
MSFT_beta

np.float64(1.1131400964401483)

Kết quả MSFT_beta = 1.113 (trong giai đoạn 2012 - 2016) có ý nghĩa rất cụ thể:

Beta = 1: Có nghĩa là cổ phiếu biến động chính xác như thị trường.

Beta < 1: Cổ phiếu ít biến động hơn thị trường (ví dụ: cổ phiếu ngành tiện ích, hàng tiêu dùng).

Beta > 1: Cổ phiếu biến động mạnh hơn thị trường.

Kết quả của bạn là 1.113, cho thấy:

Biến động mạnh hơn: Cổ phiếu Microsoft biến động mạnh hơn 11.3% so với thị trường chung (S&P 500).

Khuếch đại chuyển động:

Nếu thị trường (S&P 500) tăng 1%, giá cổ phiếu MSFT được kỳ vọng sẽ tăng 1.113%.

Ngược lại, nếu thị trường giảm 1%, giá cổ phiếu MSFT được kỳ vọng sẽ giảm 1.113%.

Mức độ rủi ro: Điều này có nghĩa là MSFT (trong giai đoạn này) được coi là một cổ phiếu có rủi ro cao hơn một chút so với mức trung bình của thị trường. Nó mang lại tiềm năng lợi nhuận cao hơn khi thị trường đi lên, nhưng cũng tiềm ẩn rủi ro thua lỗ lớn hơn khi thị trường đi xuống.